### PR TITLE
[runtime] Fix the size calculation in mono_debug_add_method ().

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/mono-debug.c [hunk 1]
```diff
@@ -461,8 +465,23 @@ mono_debug_add_method (MonoMethod *method, MonoDebugMethodJitInfo *jit, MonoDoma
...
+		/* params */
+		max_size += LEB128_MAX_SIZE;
+		max_size += jit->num_params * WRITE_VARIABLE_MAX_SIZE;
+		/* locals */
+		max_size += LEB128_MAX_SIZE;
+		max_size += jit->num_locals * WRITE_VARIABLE_MAX_SIZE;
+		/* gsharedvt */
+		max_size += 1;
+		if (jit->gsharedvt_info_var)
+			max_size += 2 * WRITE_VARIABLE_MAX_SIZE;
```
Comment (lewurm) [inline mono/metadata/mono-debug.c]: why `2`?

**Comment by chamons** [comment]
I vote this should be cherry picked to various release branches when landed.

**Comment by lewurm** [comment]
what does it fix?

**Comment by lewurm** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
The 	g_assert (size < max_size);
assertion.

**Comment by luhenry** [comment]
@monojenkins backport to 2018-06
@monojenkins backport to 2018-08
@monojenkins backport to 2018-10

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**